src/staging/streams.rs

```diff
@@ -172,24 +171,36 @@ impl<'a> Stream<'a> {
     /// Only includes ones starting from the previous minute
     pub fn arrow_files_grouped_exclude_time(
         &self,
-        exclude: NaiveDateTime,
         shutdown_signal: bool,
     ) -> HashMap<PathBuf, Vec<PathBuf>> {
+        let now = Utc::now();
+    
+        // Extract date and time components of current time
+        let now_date = (now.year(), now.month(), now.day());
+        let now_time = (now.hour(), now.minute());
+
         let mut grouped_arrow_file: HashMap<PathBuf, Vec<PathBuf>> = HashMap::new();
         let mut arrow_files = self.arrow_files();
-
-        // if the shutdown signal is false i.e. normal condition
-        // don't keep the ones for the current minute
-        if !shutdown_signal {
-            arrow_files.retain(|path| {
-                !path
-                    .file_name()
-                    .unwrap()
-                    .to_str()
-                    .unwrap()
-                    .starts_with(&exclude.format("%Y%m%dT%H%M").to_string())
-            });
-        }
+        arrow_files = arrow_files
+            .into_iter()
+            .filter(|path| {
+                let created_at = path.metadata().unwrap().created().unwrap();
```

Comment:
This will panic if run against a certain few filesystems like btrfs [ref](https://stackoverflow.com/questions/76946130/why-do-i-get-creation-time-is-not-available-on-this-platform-currently-and-wh)